cpp/src/slice2cs/CsUtil.h

```diff
@@ -6,88 +6,78 @@
-
-        // Is the mapped C# type for this field a non-nullable reference type?
-        // string fields are not included since they have a "" default.
-        static bool isMappedToRequiredField(const DataMemberPtr&);
-
-        //
-        // Generate code to marshal or unmarshal a type
-        //
-        void writeMarshalUnmarshalCode(
-            ::IceInternal::Output&,
-            const TypePtr&,
-            const std::string&,
-            const std::string&,
-            bool,
-            const std::string& = "");
-        void writeOptionalMarshalUnmarshalCode(
-            ::IceInternal::Output&,
-            const TypePtr&,
-            const std::string&,
-            const std::string&,
-            std::int32_t,
-            bool,
-            const std::string& = "");
-        void writeSequenceMarshalUnmarshalCode(
-            ::IceInternal::Output&,
-            const SequencePtr&,
-            const std::string&,
-            const std::string&,
-            bool,
-            bool,
-            const std::string& = "");
-        void writeOptionalSequenceMarshalUnmarshalCode(
-            ::IceInternal::Output&,
-            const SequencePtr&,
-            const std::string&,
-            const std::string&,
-            int,
-            bool,
-            const std::string& = "");
-    };
+    /// Convert a dimension-less array declaration to one with a dimension.
+    [[nodiscard]] std::string toArrayAlloc(const std::string& decl, const std::string& size);
+
+    /// Returns the namespace of a Contained entity.
+    [[nodiscard]] std::string getNamespace(const ContainedPtr& p);
+
+    [[nodiscard]] std::string getUnqualified(const ContainedPtr& p, const std::string& package);
+
+    /// Removes a leading '@' character from the provided identifier (if one is present).
+    [[nodiscard]] std::string removeEscapePrefix(const std::string& identifier);
+
+    [[nodiscard]] std::string typeToString(const TypePtr& type, const std::string& package, bool optional = false);
+
+    /// Returns the namespace prefix of a Contained entity.
+    [[nodiscard]] std::string getNamespacePrefix(const ContainedPtr& p);
+
+    [[nodiscard]] std::string
+    resultStructName(const std::string& className, const std::string& opName, bool marshaledResult = false);
+    [[nodiscard]] std::string resultType(const OperationPtr& op, const std::string& package, bool dispatch = false);
+    [[nodiscard]] std::string taskResultType(const OperationPtr& op, const std::string& scope, bool dispatch = false);
+    [[nodiscard]] std::string getOptionalFormat(const TypePtr& type);
+    [[nodiscard]] std::string getStaticId(const TypePtr& type);
+
+    /// Is this Slice type mapped to a C# value type?
+    [[nodiscard]] bool isValueType(const TypePtr& type);
+
+    /// Is this Slice struct mapped to a C# class?
+    [[nodiscard]] inline bool isMappedToClass(const StructPtr& p) { return !isValueType(p); }
```

Comment:
The one exception is that I `inline`d this function, since it's very short and no longer a member function.